server/githookHandler.go

```diff
@@ -17,30 +18,43 @@ import (
 	"github.com/google/uuid"
 )
 
-// TransformListen converts a Github event into an Eventgrid event and publishes the event to Eventgrid
-func TransformListen(c buffalo.Context) error {
-
+// GithookReceiver converts a Github event into an Eventgrid event and publishes the event to Eventgrid
+func GithookReceiver(c buffalo.Context) error {
+	webhookSecret := os.Getenv("webhookSecret");
+	eventgridSecret := os.Getenv("eventgridSecret");
+	eventgridEndpoint := os.Getenv("eventgridEndpoint");
+	
 	request := c.Request()
-	// payload, err := github.ValidatePayload(request, []byte(os.Getenv("APPSETTING_X_HUB_SIGNATURE")))
-	// if err != nil {
-	// 	log.Printf("secret key is not correct: err=%s\n", err)
-	// 	return c.Error(http.StatusInternalServerError, err)
-	// }
-	var err error
-	var body []byte
-	if body, err = ioutil.ReadAll(request.Body); err != nil {
-		return err
+
+	// save request body so it can be read multiple times
+	var bodyBytes []byte
+
+	if request.Body != nil {
+		bodyBytes, _ = ioutil.ReadAll(request.Body)
```

Comment:
In general, this is an attack vector against Go programs. Given that we're doing a big overhaul, it may be worth using a [`LimitReader`](https://godoc.org/io#LimitedReader) instead. I know that GitHub guarantees a 5MB limit or something anyway, so we might as well clamp to that. Maybe inject that limit with a default of 5MB?

Reply:
25MB yeh sgtm